Es output docs (#7)
logstash-plugins/logstash-input-elastic_serverless_forwarder: docs/index.asciidoc

NOTE: Basic Authentication is not a substitute for SSL, as it provides neither secrecy nor security on its own.
      When used with SSL disabled, HTTP Basic credentials are transmitted in effectively clear-text and can be easily recovered by an adversary.

[id="plugins-{type}s-{plugin}-es-output-notes"]
==== Using {esf-name} with the Elasticsearch output
Here are some tips for configuring the {esf} input to work with the elasticsearch output:

* Set the `document_id` in the output configuration when you use the {esf}  input with an {logstash-ref}/plugins-outputs-elasticsearch.html#plugins-outputs-elasticsearch[Elasticsearch output plugin].
+
[source,ruby]
----
output {
  elasticsearch {
      ...
      document_id => "%{[@metadata][_id]}"
      ...
  }
}
----

* For {esf-name} v1.9.0 and earlier, rename the field `_id` to `@metadata._id` with a filter:
+
[source,ruby]
----
filter {
    # support ESF < 1.10
    if [_id] and ![@metadata][_id] {
      mutate { rename => { "_id" => "[@metadata][_id]" } }
    }
}
----

> * Starting from version 1.10.0 of {esf-name}, the `_id` field is no longer available, and instead, Logstash now receives the `@metadata._id` field. Configuring `document_id` as shown in the example above is sufficient.

Reviewer: nit: 
```
Configuring `document_id` as shown in the example above is sufficient.
```

the example just above is the filter, maybe?
```
Configuring `document_id` as shown in the first code example is sufficient.
```

Reviewer: or we flip `1.10` and `<=1.9`

@karenzone what do you think?

Reviewer 2: @aspacca, either approach works so there's no wrong answer here.  But given a choice, I prefer  to " flip 1.10 and <=1.9".  We always encourage users to use the latest versions, and we should make that info higher in the doc and easier to find rather than making them keep reading to find relevant info.

Reviewer: @karenzone , please, have a look to the changes. thanks :)

NOTE: Basic Authentication is not a substitute for SSL, as it provides neither secrecy nor security on its own.
      When used with SSL disabled, HTTP Basic credentials are transmitted in effectively clear-text and can be easily recovered by an adversary.

[id="plugins-{type}s-{plugin}-es-output-notes"]
==== Using {esf-name} with the Elasticsearch output
Here are some tips for configuring the {esf} input to work with the elasticsearch output:

* Set the `document_id` in the output configuration when you use the {esf}  input with an {logstash-ref}/plugins-outputs-elasticsearch.html#plugins-outputs-elasticsearch[Elasticsearch output plugin].

> 

Reviewer 2: ```suggestion
+
```

[source,ruby]
----
output {
  elasticsearch {
      ...
      document_id => "%{[@metadata][_id]}"
      ...
  }
}
----

* For {esf-name} v1.9.0 and earlier, rename the field `_id` to `@metadata._id` with a filter:
+

> 

Reviewer 2: ```suggestion
```

include::{include_path}/{type}.asciidoc[]

:default_codec!:

==== Elasticsearch output
When this input is used along with an {logstash-ref}/plugins-outputs-elasticsearch.html#plugins-outputs-elasticsearch[Elasticsearch output pluginh], you have to set the `document_id` in the output configuration.

[source]
----
output {
  elasticsearch {
      ...
      document_id => "%{[@metadata][_id]}"
      ...
  }
}
----

* For {esf-name} v1.9.0 and below you have to rename the field `_id` to `@metadata._id` with a filter:

> [source]

Reviewer 3: I believe ASCIIDOC needs a blank line before this block:

```suggestion

[source]
```

----

> filter {
>     mutate { rename => { "_id" => "[@metadata][_id]" } }
> }

Reviewer 3: While mutate's `rename` operation _effectively_ becomes a no-op for a source field that isn't present (such as an `_id` that is _not_ present on ESF 1.10), it may be useful to suggest a conditional wrapper that only runs the filter if `[_id]` is present and `[@metadata][_id]` is _not_ present to avoid ambiguity in cases where both are present.

The following should work with events sent by the old 1.9.x while being fully-forward-compatible with 1.10 where the rename isn't needed:

```suggestion
filter {
    # support ESF < 1.10
    if [_id] and ![@metadata][_id] {
      mutate { rename => { "_id" => "[@metadata][_id]" } }
    }
}
```

----

> * Snce v1.10.0 of {esf-name}, the field `_id` is not present anymore and the `@metadata._id` one is s now sent to Logstash: only setting `document_id` as in the sample configuration above is enough.

Reviewer 4: ```suggestion
* Starting from version 1.10.0 of {esf-name}, the `_id` field is no longer available, and instead, Logstash now receives the `@metadata._id` field. Simply configuring `document_id` as shown in the example above is sufficient."
```

include::{include_path}/{type}.asciidoc[]

:default_codec!:

> ==== Elasticsearch output

Reviewer 2: ```suggestion
[id="plugins-{type}s-{plugin}-es-output-notes"]
==== Using {esf-name} with the Elasticsearch output
```

Reviewer 2: Suggest adding more words for better SEO performance and context in search results

> ==== Elasticsearch output

Reviewer 2: Please relocate this helpful info to around Line 134.  It's getting lost behind the Configuration and Common options. 

`:default_codec!:` should be the last line in the file.

> When this input is used along with an {logstash-ref}/plugins-outputs-elasticsearch.html#plugins-outputs-elasticsearch[Elasticsearch output pluginh], you have to set the `document_id` in the output configuration.

Reviewer 2: ```suggestion
Here are some tips for configuring the {esf} input to work with the elasticsearch output: 

* Set the `document_id` in the output configuration when you use the {esf}  input with an {logstash-ref}/plugins-outputs-elasticsearch.html#plugins-outputs-elasticsearch[Elasticsearch output plugin].
```
+

> [source]

Reviewer 2: ```suggestion
[source,ruby]
```

----
output {
  elasticsearch {
      ...
      document_id => "%{[@metadata][_id]}"
      ...
  }
}
----

> * For {esf-name} v1.9.0 and below you have to rename the field `_id` to `@metadata._id` with a filter:

Reviewer 2: ```suggestion
* For {esf-name} v1.9.0 and earlier, rename the field `_id` to `@metadata._id` with a filter:
+
```

> [source]

Reviewer 2: ```suggestion
[source,ruby]
```

----
filter {
    # support ESF < 1.10
    if [_id] and ![@metadata][_id] {
      mutate { rename => { "_id" => "[@metadata][_id]" } }
    }
}
----

> * Starting from version 1.10.0 of {esf-name}, the `_id` field is no longer available, and instead, Logstash now receives the `@metadata._id` field. Simply configuring `document_id` as shown in the example above is sufficient."

Reviewer 2: ```suggestion
* Starting from version 1.10.0 of {esf-name}, the `_id` field is no longer available, and instead, Logstash now receives the `@metadata._id` field. Configuring `document_id` as shown in the example above is sufficient.
```
